.github/actions/set-env-variables/action.yml

```diff
@@ -16,7 +16,22 @@ runs:
         echo "CILIUM_CLI_VERSION=$CILIUM_CLI_VERSION" >> $GITHUB_ENV
         echo "CILIUM_CLI_IMAGE_REPO=quay.io/cilium/cilium-cli-ci" >> $GITHUB_ENV
         echo "CILIUM_CLI_SKIP_BUILD=true" >> $GITHUB_ENV
-        echo "CILIUM_CLI_CODE_OWNERS_PATHS=CODEOWNERS" >> $GITHUB_ENV
+
+        # Detect if we're on a stable branch (vX.Y pattern) and include TESTOWNERS if available
+        # To create TESTOWNERS on a stable branch: cp CODEOWNERS TESTOWNERS && git add TESTOWNERS && git commit
+        BRANCH_NAME="${GITHUB_REF_NAME:-$(git rev-parse --abbrev-ref HEAD)}"
+        if [[ "$BRANCH_NAME" =~ ^v[0-9]+\.[0-9]+$ ]]; then
+          # On a stable branch, use TESTOWNERS if it exists, otherwise fall back to CODEOWNERS
+          if [[ -f "TESTOWNERS" ]]; then
+            echo "CILIUM_CLI_CODE_OWNERS_PATHS=TESTOWNERS,CODEOWNERS" >> $GITHUB_ENV
+          else
+            echo "CILIUM_CLI_CODE_OWNERS_PATHS=CODEOWNERS" >> $GITHUB_ENV
+          fi
+        else
+          # On main or other branches, use CODEOWNERS only
+          echo "CILIUM_CLI_CODE_OWNERS_PATHS=CODEOWNERS" >> $GITHUB_ENV
+        fi
```

Comment:
I wonder if a more generic solution might just be to declare the variable initially only locally with the path CODEOWNERS, then add a file exists check to add the TESTOWNERS file, and iff it exists, add that path to the variable. Then finally declare the variable by emitting it to `$GITHUB_ENV`.